### PR TITLE
Make LedgerMetadata fields for closing immutable

### PR DESCRIPTION
**Pull request by ivankelly**
Master issue: #281

### CODE REVIEW

**Comment by ivankelly** [comment]
I need to fix up some tests, they seem to be making assumptions about unclosed ledgers

**Comment by ivankelly** [comment]
I need to fix up some tests, they seem to be making assumptions about unclosed ledgers

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadataBuilder.java [hunk 1]
```diff
@@ -73,14 +73,9 @@ public static LedgerMetadataBuilder from(LedgerMetadata other) {
...
         builder.state = other.getState();
-
-        long lastEntryId = other.getLastEntryId();
-        if (lastEntryId != LedgerHandle.INVALID_ENTRY_ID) {
-            builder.lastEntryId = Optional.of(lastEntryId);
-        }
-        long length = other.getLength();
-        if (length > 0) {
-            builder.length = Optional.of(length);
+        if (builder.state == LedgerMetadataFormat.State.CLOSED) {
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadataBuilder.java]: Is this a behavior change?
Do you think it is worth to add a comment?

Comment (ivankelly) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerMetadataBuilder.java]: The previous behaviour was incorrect. length should never be set if the state is anything but closed. I don't think it needs a comment. the intent is clear.

**Pull request merged**